src/bin.ts

```diff
@@ -180,75 +184,83 @@ const deploy = async ({ yes, bucket }: { yes: boolean, bucket: string }) => {
         spinner.text = 'Syncing...';
         const publicDir = resolve('./public');
         const stream = klaw(publicDir);
-        const promises: Promise<any>[] = [];
         let isKeyInUse: { [objectKey: string]: boolean } = {};
 
         stream.on('data', async ({ path, stats }) => {
             if (!stats.isFile()) {
                 return;
             }
-
-            const key = createSafeS3Key(relative(publicDir, path));
-            const buffer = await readFile(path);
-            const tag = `"${createHash('md5').update(buffer).digest('hex')}"`;
-            const object = objects.find(object => object.Key === key && object.ETag === tag);
-
-            isKeyInUse[key] = true;
-
-            if (object) {
-                // object with exact hash already exists, abort.
-                return;
-            }
-
-            try {
-                const promise = new S3.ManagedUpload({
-                    service: s3,
-                    params: {
-                        Bucket: config.bucketName,
-                        Key: key,
-                        Body: fs.createReadStream(path),
-                        ACL: config.acl === null ? undefined : (config.acl || 'public-read'),
-                        ContentType: mime.getType(path) || 'application/octet-stream',
-                        ...getParams(key, params)
+            uploadQueue.push(async (callback: () => void) => {
```

Comment:
Having an ES2017 async function with a Node-style callback argument is a bit confusing. According to the [async docs](https://caolan.github.io/async/global.html) you don't need to use the callback argument, you can just return the value like normal in the async function. If you tried this and it didn't work, it's most likely because we currently transpile our async/await functions due to #60.

Until we act on #60, I suggest you use [asyncify](https://caolan.github.io/async/docs.html#asyncify) to work around this.